main.py

```diff
@@ -46,19 +62,80 @@ def handler(event, context):
 
 def configure_confluent_kafka_consumer(event, args):
     if 'AWS_PROFILE' in os.environ:
-        boto3.setup_default_session(profile_name=args.aws_profile,
-                                    region_name=args.aws_region)
+     boto3.setup_default_session(profile_name=args.aws_profile,
+     region_name=args.aws_region)
 
     if logger.isEnabledFor(logging.DEBUG):
         # Log everything from boto3
         boto3.set_stream_logger()
-        logger.debug(f"Using boto3 {boto3.__version__}")
+    logger.debug(f"Using boto3 {boto3.__version__}")
 
     logger.debug(event)
 
     message = json.loads(event['Records'][0]['Sns']['Message'])
     logger.debug(message)
 
+    private_ip = message['detail']['containers'][0]['networkInterfaces'][0]['privateIpv4Address']
+    logger.debug(private_ip)
+
+    payload = {
+        "config": {
+            "connector.name": args.connector_name,
+            "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+            "tasks.max": args.tasks_max,
+            "topics": args.topics,
+            "flush.size": args.flush_size,
+            "s3.region": args.aws_region,
+            "s3.bucket.name": args.s3_bucket_name,
+            "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+            "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
+            "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
+            "partitioner.class": "io.confluent.connect.storage.partitioner.DefaultPartitioner",
+        }
+    }
+
+
+# Get a list of existing connectors
+
+    response = requests.get(f"http://{private_ip}:{args.port}/connectors")
+    existing_connectors = response.content.decode("utf-8")
+    logger.debug(existing_connectors)
+
+    # PUT payload if connectors do exist this updates existing connector
+
+    if args.connector_name in existing_connectors:
+        logger.debug("update connector [PUT]")
+
+        payload = {
+            "config": {
+                "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+                "tasks.max": args.tasks_max,
+                "topics": args.topics,
+                "flush_size": args.flush_size,
+
+            }
+        }
+
+        response = requests.put(f"http://{private_ip}:{args.port}/connectors/{args.connector_name}/config", json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+    else:
+        # POST payload if connectors don't exist
+
+        logger.debug("create connector [POST]")
+
+        response = requests.post(f"http://{private_ip}:{args.port}/connectors", json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+        # DELETE all others
+
+    for existing_connector in existing_connectors:
+
+        logger.debug("delete connector [DELETE]")
+
+        response = requests.delete(f"http://{private_ip}:{args.port}/connectors/{existing_connector}")
```

Comment:
I suspect `existing_connector` is a Python object/dict, so you'll have to reference one of its attributes.  Now you've got me wondering whether `existing_connectors` is even iterable...are you able to run this code locally and run it through a debugger to see what both of those objects look like?

Reply:
I have already run this, existing connectors is a list of strings, so both this line and the for loop are valid